src/main/java/com/microsoft/sqlserver/jdbc/SQLServerConnection.java

```diff
@@ -122,42 +123,41 @@
     
     private Boolean isAzureDW = null;
 
-    static class Sha1HashKey implements java.io.Serializable {
+    static class CityHash128Key implements java.io.Serializable {
 
         /**
          * Always refresh SerialVersionUID when prompted
          */
         private static final long serialVersionUID = 166788428640603097L;
-        private byte[] bytes;
+        String unhashedString;
+        private long[] segments;
+        private int hashCode;
 
-        Sha1HashKey(String sql,
+        CityHash128Key(String sql,
                 String parametersDefinition) {
-            this(String.format("%s%s", sql, parametersDefinition));
+            this(sql + parametersDefinition);
         }
 
-        Sha1HashKey(String s) {
-            bytes = getSha1Digest().digest(s.getBytes());
+        CityHash128Key(String s) {
+            unhashedString = s;
+            byte[] bytes = new byte[s.length()];
+            s.getBytes(0, s.length(), bytes, 0);
```

Comment:
getBytes is a deprecated method - suppress warning or replace?

Reply:
Suppress the warning -- there is an excellent performance reason why we're using it. And as Brett has pointed out, it's never going away, because it's sometimes the right thing to do.